docs/proposals/platform-extension-implementation-extension-operators.md

```diff
@@ -0,0 +1,86 @@
+# Platform Extension Implementation: Extension Operators
+
+**Author**: Moritz Bracht
+
+**Status**: Draft proposal.
+
+## Goals
+
+This proposal explores an operator based approach for KKP extensions.
+
+## Motivation and Background
+
+An implementation of platform-extension based on Helm charts provides flexible
+yet simplistic way to manage the lifecycle of platform-extensions.
+
+In case of more complex extensions that can not simply be installed with Helm
+([like KubeVirt](https://kubevirt.io/user-guide/operations/installation/), the
+extensions could be managed by their own operators.
+
+To cover simple Helm based extensions either a Helm operator could be installed
+or simple operators could be [generated directly from Helm
+charts](https://sdk.operatorframework.io/docs/building-operators/helm/tutorial/).
+
+[Check this for requirements](./platform-extensions.md#requirements-for-implementations)
+
+## Implementation
+
+Every extension is managed by its own operator. These operators are being
+managed by [OLM](https://olm.operatorframework.io/).
+
+### Main Concept Points
+
+* Extensions provided by KKP devs
+  * Extension operators are bundled and provided via OLM catalog
```

Comment:
Small hint on how the OLM catalog is realized needed. So that a reader has not dive into OLM in the first step.

Reply:
Sure, I will extend this section and also the one of the Extension Catalog Controller.